app/assets/javascripts/misc/personal-key-page-controller.js

```diff
@@ -35,19 +35,14 @@ function resetForm() {
 function handleSubmit(event) {
   event.preventDefault();
 
-  const invalidMatches = inputs.reduce(function(accumulator, input, index) {
-    const value = input.value;
+  const value = input.value;
+  const key = [];
 
-    if (value.toUpperCase() === personalKeyWords[index].innerHTML.replace(/\s+/, '').toUpperCase()) {
-      return accumulator;
-    }
-
-    accumulator.push(value);
-
-    return accumulator;
-  }, []);
+  personalKeyWords.forEach((group) => {
```

Comment:
Since the personal key that we are comparing the user's input against never changes, this code and the `key.join('-').toUpperCase()` could just be done once at the beginning of the file.